Android Auto map rotates and twitches unpredictably when stopped

⚠ Have you searched for similar, already existing issues?
Yes. 

**Describe the issue**
When I have Organic Maps on my car's screen through Android Auto, and stop near an intersection, the map frequently starts rotating, or moving in unpredictable directions (videos attached). Sometimes, the movement is quite subtle, sometimes, the entire map rotates back and forth more than 45 degrees from the direction the car is pointed in. While I was using my phone running Organic Maps to record the videos attached to this issue, the random twitches to the map view happen even when the phone is sitting in the center console and not moving. This issue has been around ever since I started using Organic Maps more than a year ago.


**Steps to reproduce**
1. Connect your phone to Android Auto, start navigation
2. Stop at an intersection
3. Watch the map rotate and twitch unpredictably 


**Expected behaviour**
When the car is not moving, the map view should not move either.

**Screenshots**

https://github.com/user-attachments/assets/3c7c9412-fddb-4267-971e-11790d087721
https://github.com/user-attachments/assets/ba0288db-c66f-46f6-ac4c-aa2d39deb12b

**System information:**
 - Operating system and its version: [GrapheneOS - Android 14, Android 15]
 - Organic Maps version: [All versions since ~January 2024 – 2024.11.27-12]
 - Device Model: [Pixel 8 Pro]


**Additional context**
I can privately send more recordings, as well as the model of my car.

The sideways twitching issue seems to become especially noticeable after I don't take a turn OM navigation wanted me to take.

https://github.com/user-attachments/assets/cdc5eeb2-e18c-4e0f-b8e2-931c7e464d69

```
type: logcat
osVersion: google/husky/husky:15/AP4A.250205.002/2025030200:user/release-keys
flags: dev options enabled
package: app.organicmaps:25021703, targetSdk 35
buffers: main,system,crash,events,kernel
level: verbose

--------- beginning of crash
03-03 18:23:38.724 32631 32631 E AndroidRuntime: FATAL EXCEPTION: main
03-03 18:23:38.724 32631 32631 E AndroidRuntime: Process: app.organicmaps, PID: 32631
03-03 18:23:38.724 32631 32631 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@1b191dc with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{9c45631 32631:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{9c45631 32631:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 18:23:38.724 32631 32631 E AndroidRuntime: 	... 11 more
03-03 19:47:36.587 27063 27063 E AndroidRuntime: FATAL EXCEPTION: main
03-03 19:47:36.587 27063 27063 E AndroidRuntime: Process: app.organicmaps, PID: 27063
03-03 19:47:36.587 27063 27063 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@a7e0bfa with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{3536ede 27063:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{3536ede 27063:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 19:47:36.587 27063 27063 E AndroidRuntime: 	... 11 more
03-03 20:23:53.691  8078  8078 E AndroidRuntime: FATAL EXCEPTION: main
03-03 20:23:53.691  8078  8078 E AndroidRuntime: Process: app.organicmaps, PID: 8078
03-03 20:23:53.691  8078  8078 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@c28f7b8 with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{e800f40 8078:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{e800f40 8078:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 20:23:53.691  8078  8078 E AndroidRuntime: 	... 11 more
03-03 21:48:34.978 16337 16337 E AndroidRuntime: FATAL EXCEPTION: main
03-03 21:48:34.978 16337 16337 E AndroidRuntime: Process: app.organicmaps, PID: 16337
03-03 21:48:34.978 16337 16337 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@c652631 with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{420eb7f 16337:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{420eb7f 16337:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 21:48:34.978 16337 16337 E AndroidRuntime: 	... 11 more
03-03 22:06:29.455 29417 29417 E AndroidRuntime: FATAL EXCEPTION: main
03-03 22:06:29.455 29417 29417 E AndroidRuntime: Process: app.organicmaps, PID: 29417
03-03 22:06:29.455 29417 29417 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@c887db4 with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{238227c 29417:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{238227c 29417:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 22:06:29.455 29417 29417 E AndroidRuntime: 	... 11 more
03-03 23:18:18.547 20247 20247 E AndroidRuntime: FATAL EXCEPTION: main
03-03 23:18:18.547 20247 20247 E AndroidRuntime: Process: app.organicmaps, PID: 20247
03-03 23:18:18.547 20247 20247 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@a44576b with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{d378f58 20247:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{d378f58 20247:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-03 23:18:18.547 20247 20247 E AndroidRuntime: 	... 11 more
03-04 16:56:14.396 24618 24618 E AndroidRuntime: FATAL EXCEPTION: main
03-04 16:56:14.396 24618 24618 E AndroidRuntime: Process: app.organicmaps, PID: 24618
03-04 16:56:14.396 24618 24618 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@d80a44f with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{bec12fc 24618:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{bec12fc 24618:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-04 16:56:14.396 24618 24618 E AndroidRuntime: 	... 11 more
03-04 17:55:00.768 31567 31567 E AndroidRuntime: FATAL EXCEPTION: main
03-04 17:55:00.768 31567 31567 E AndroidRuntime: Process: app.organicmaps, PID: 31567
03-04 17:55:00.768 31567 31567 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@304e15d with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{8d78fb 31567:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{8d78fb 31567:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-04 17:55:00.768 31567 31567 E AndroidRuntime: 	... 11 more
03-04 18:11:50.531 16379 16379 E AndroidRuntime: FATAL EXCEPTION: main
03-04 18:11:50.531 16379 16379 E AndroidRuntime: Process: app.organicmaps, PID: 16379
03-04 18:11:50.531 16379 16379 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@4638ffd with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{74ac775 16379:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{74ac775 16379:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-04 18:11:50.531 16379 16379 E AndroidRuntime: 	... 11 more
03-04 18:43:30.688 30066 30066 E AndroidRuntime: FATAL EXCEPTION: main
03-04 18:43:30.688 30066 30066 E AndroidRuntime: Process: app.organicmaps, PID: 30066
03-04 18:43:30.688 30066 30066 E AndroidRuntime: java.lang.RuntimeException: Unable to start service app.organicmaps.routing.NavigationService@34424d9 with Intent { cmp=app.organicmaps/.routing.NavigationService }: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b0e0021 30066:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5280)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2555)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8826)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{b0e0021 30066:app.organicmaps/u0a207} targetSDK=35 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3240)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3224)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3200)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3142)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7231)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:778)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at app.organicmaps.routing.NavigationService.onStartCommand(NavigationService.java:231)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5262)
03-04 18:43:30.688 30066 30066 E AndroidRuntime: 	... 11 more
--------- beginning of events
03-04 19:18:45.577  1444  1444 I wm_on_destroy_called: [Token=244259912,Component Name=app.organicmaps.MapPlaceholderActivity,Reason=performDestroy,time=22ms]

```

--- a/android/app/src/main/java/app/organicmaps/location/LocationHelper.java
+++ b/android/app/src/main/java/app/organicmaps/location/LocationHelper.java
@@ -213,7 +213,7 @@ public class LocationHelper implements BaseLocationProvider.Listener
 
     if (mSavedLocation != null)
     {
-      if (!LocationUtils.isFromFusedProvider(location) && !LocationUtils.isLocationBetterThanLast(location, mSavedLocation))
+      if (!LocationUtils.isLocationBetterThanLast(location, mSavedLocation))
       {
         Logger.d(TAG, "The new " + location + " is worse than the last " + mSavedLocation);
         return;

--- a/android/app/src/main/java/app/organicmaps/util/LocationUtils.java
+++ b/android/app/src/main/java/app/organicmaps/util/LocationUtils.java
@@ -14,14 +14,11 @@ import android.view.Surface;
 import androidx.annotation.NonNull;
 import androidx.core.content.ContextCompat;
 
-import java.util.Objects;
-
 public class LocationUtils
 {
   private LocationUtils() {}
 
   private static final double DEFAULT_SPEED_MPS = 5;
-  private static final float THRESHOLD_DISTNACE_METERS = 3; //meters
   public static final String FUSED_PROVIDER = "fused";
 
   /**
@@ -87,8 +84,7 @@ public class LocationUtils
 
   public static boolean isLocationBetterThanLast(@NonNull Location newLocation, @NonNull Location lastLocation)
   {
-    //if provider is changing but distance between the two co-ords are less than threshold then discard
-    if (!Objects.equals(lastLocation.getProvider(), newLocation.getProvider()) && lastLocation.distanceTo(newLocation) < THRESHOLD_DISTNACE_METERS)
+    if (newLocation.getElapsedRealtimeNanos() < lastLocation.getElapsedRealtimeNanos())
       return false;
 
     // As described in isAccuracySatisfied, GPS may have zero accuracy "for some reasons".

--- a/android/app/src/main/java/app/organicmaps/util/LocationUtils.java
+++ b/android/app/src/main/java/app/organicmaps/util/LocationUtils.java
@@ -14,11 +14,14 @@ import android.view.Surface;
 import androidx.annotation.NonNull;
 import androidx.core.content.ContextCompat;
 
+import java.util.Objects;
+
 public class LocationUtils
 {
   private LocationUtils() {}
 
   private static final double DEFAULT_SPEED_MPS = 5;
+  private static final float THRESHOLD_DISTNACE_METERS = 3; //meters
   public static final String FUSED_PROVIDER = "fused";
 
   /**
@@ -84,6 +87,10 @@ public class LocationUtils
 
   public static boolean isLocationBetterThanLast(@NonNull Location newLocation, @NonNull Location lastLocation)
   {
+    //if provider is changing but distance between the two co-ords are less than threshold then discard
+    if (!Objects.equals(lastLocation.getProvider(), newLocation.getProvider()) && lastLocation.distanceTo(newLocation) < THRESHOLD_DISTNACE_METERS)
+      return false;
+
     // As described in isAccuracySatisfied, GPS may have zero accuracy "for some reasons".
     if (isFromGpsProvider(lastLocation) && lastLocation.getAccuracy() == 0.0f)
       return true;

